### DISCUSSION TITLE
Better `ClassDB::register_class` documentation and variation

### DISCUSSION THREAD

**Discussion by Withaust** [godotengine/godot-proposals#10558] [Scripting]
Currently, we only got access to these templates:
```cpp

template <typename T>
void ClassDB::register_class(bool p_virtual) {
    ClassDB::_register_class<T, false>(p_virtual);
}

template <typename T>
void ClassDB::register_abstract_class() {
    ClassDB::_register_class<T, true>();
}

template <typename T>
void ClassDB::register_internal_class() {
    ClassDB::_register_class<T, false>(false, false);
}

template <typename T>
void ClassDB::register_runtime_class() {
    ClassDB::_register_class<T, false>(false, true, true);
}
```

My main issue is that its not really apparent what these distinct 4 are used for in context of GDExtension <=> GDScript binding, while also missing some of the options like runtime virtual class registration.

**Comment by dsnopek** [comment]
I agree that it'd be good to document these. However, we should document the macros (ie `GDREGISTER_CLASS()`, `GDREGISTER_VIRTUAL_CLASS()`, etc) since those are the recommended way to register classes.
